File: lab/python/textos/vogais_em_uma_palavra.py
import re


def contagem_de_vogais(texto):
    vogais = "aeiouãẽĩõũáéíóúàèìòùâêîôû"
    contador = 0
    for palavra in texto.lower():
        if palavra in vogais:
            contador += 1
    return contador


def contagem_vogais_versao_regex(texto):
    # O padrão abaixo procura qualquer vogal (maiúscula ou minúscula)
    padrao = r"[aeiouãẽĩõũáéíóúàèìòùâêîôûAEIOUÃẼĨÕŨÁÉÍÓÚÀÈÌÒÙÂÊÎÔÛ]"
    vogais = re.findall(padrao, texto)
    return len(vogais)


print(contagem_de_vogais('Programação'))
print(contagem_vogais_versao_regex('Programação'))

File: lab/c/textos/vogais_em_uma_palavra.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int contagem_de_vogais(const char *texto) {
    const char *vogais = "aeiouAEIOUãẽĩõũáéíóúàèìòùâêîôûÃẼĨÕŨÁÉÍÓÚÀÈÌÒÙÂÊÎÔÛ";
    int contador = 0;
    for (int i = 0; texto[i] != '\0'; i++) {
        if (strchr(vogais, texto[i])) {
            contador++;
        }
    }
    return contador;
}

int main(void) {
    const char *texto = "Programação";
    printf("%d\n", contagem_de_vogais(texto)); // 5
    printf("%d\n", contagem_de_vogais(texto)); // mesmo resultado
    return 0;
}
